tests/testing_common.py

```diff
@@ -615,9 +656,12 @@ def _test_merge_layers_nan(self, model_id, config_cls, config_kwargs):
             base_model_name_or_path=model_id,
             **config_kwargs,
         )
+
         model = get_peft_model(model, config)
         model = model.to(self.torch_device)
 
+        self.perturb_trainable_token_weights_if_used(model, config_kwargs)
```

Comment:
Ah, a bit unfortunate that this is necessary. WDYT about using an auto fixture that monkey patches `TrainableTokensLayer.update_layer` so that it calls "super" but then perturbs the weight?

Reply:
I think that something that changes weights should be plainly visible, even though it is not doing anything 80% of the time. If we change something about this arrangement, then it should focus on getting `init_weights` to work with trainable tokens IMO (or something else that involves less implicit magic than there already is).

Reply:
I can see that argument. Fine with leaving it as is.